Add -skipcrossarchnative and -verbose options

giuliobenetti opened the pull request:
Build.sh lacks -skipcrossarchnative and -verbose for coreclr building so let's add these 2 options to be forwarded to coreclr build system.

giuliobenetti committed:
Add CoreclrVerbose=true when --verbose is passed as option to build.sh

Sponsored by: Tekvox Inc.
Cc: Jim Reinhart <jimr@tekvox.com>
Cc: James Autry <jautry@tekvox.com>
Cc: Matthew Maron <matthewm@tekvox.com>
Signed-off-by: Giulio Benetti <giulio.benetti+tekvox@benettiengineering.com>

giuliobenetti commented:
Kindly ping this PR. Do the patches look good?

hoyosjs commented on eng/build.sh:
```diff
@@ -524,6 +524,16 @@ while [[ $# > 0 ]]; do
       shift 2
       ;;
 
+      -skipcrossarchnative)
```
This argument no longer does anything. It might be better to remove it from the help.

giuliobenetti replied:
@hoyosjs thanks for reviewing. You're right, but I also found a left-over here:
https://github.com/dotnet/runtime/blob/main/src/coreclr/build-runtime.sh#L23
I can add a patch to remove it in this PR in place of the skipcrossarchnative patch Is it ok for your?

hoyosjs commented on eng/build.sh:
```diff
@@ -524,6 +524,16 @@ while [[ $# > 0 ]]; do
+      ;;
+
+      -verbosity)
+      arguments="$arguments /p:Verbose=true"
```
hm, the parameter should be verbose, to match the native script. Also, the transport property name (Verbose) is a very common name - maybe call it CoreclrVerbose until others start using it?

giuliobenetti replied:
Hi @hoyosjs,
I've just updated this commit, removed the other commit about -skipcrossarchnative options and added a commit to drop comment about -skipcrossarchnative.

giuliobenetti committed:
Remove leftover comment about -skipcrossarchnative

During commit 7d562f9d2a4285881ae1c412aecb164dc9370013 comment about
-skipcrossarchnative has been forgotten, so let's remove it since that
flag is not available anymore.

Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>

hoyosjs commented on eng/build.sh:
```diff
@@ -524,6 +524,11 @@ while [[ $# > 0 ]]; do
 
+      -verbose)
+      arguments="$arguments /p:CoreclrVerbose=true"
+      shift 2
```
```suggestion
      shift 1
```

hoyosjs committed:
Correct shift value

giuliobenetti commented on eng/build.sh:
```diff
@@ -526,7 +526,7 @@ while [[ $# > 0 ]]; do
       -verbose)
       arguments="$arguments /p:CoreclrVerbose=true"
-      shift 2
+      shift 1
```
I've missed this, thank you!
